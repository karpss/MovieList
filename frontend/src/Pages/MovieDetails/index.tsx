import React from 'react';
import { useQuery } from '@tanstack/react-query';
import { Link, useParams } from 'react-router-dom';
import { Typography } from '@mui/material';
import { getMovieDetails } from '../../api/movies';

type MovieDetailsParams = {
  movieName: string;
};

function MovieDetails() {
  const { movieName } = useParams<
    keyof MovieDetailsParams
  >() as MovieDetailsParams;
  const {
    status: movieStatus,
    error: movieError,
    data: movie,
  } = useQuery({
    queryKey: ['movies', movieName],
    queryFn: () => getMovieDetails(movieName),
  });

  if (movieStatus === 'loading') return <h1>Loading...</h1>;

  if (movieStatus === 'error') {
    return <h1>{JSON.stringify(movieError)}</h1>;
  }

  return (
    <div data-testid="MovieDetails">
      <Typography variant="h2">
        <Link to="/">All Movies</Link>
      </Typography>
      <Typography variant="h3" gutterBottom>
        {movie.movieName}
      </Typography>

      <Typography variant="subtitle1" gutterBottom>
        {movie.releaseDate}
      </Typography>
    </div>
  );
}
export default MovieDetails;
