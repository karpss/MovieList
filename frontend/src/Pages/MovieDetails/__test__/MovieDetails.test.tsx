import { render } from '@testing-library/react';
import React from 'react';
import { describe, it, expect } from 'vitest';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import MovieDetails from '..';

describe('Movie Details', () => {
  const queryClient = new QueryClient();
  it('renders', () => {
    const wrapper = render(
      <QueryClientProvider client={queryClient}>
        <MovieDetails />
      </QueryClientProvider>
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
