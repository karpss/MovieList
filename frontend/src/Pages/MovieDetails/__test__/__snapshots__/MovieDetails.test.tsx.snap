// Vitest Snapshot v1

exports[`Movie Details > renders 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;
